src/NServiceBus.Core/Correlation/CorrelationContextExtensions.cs

```diff
@@ -10,6 +10,10 @@ public static class CorrelationContextExtensions
         /// </summary>
         /// <param name="options">Options being extended.</param>
         /// <param name="correlationId">The custom correlation id.</param>
+        [ObsoleteEx(
+            Message = "Setting a custom correlation ID is discouraged and will be removed in the next major version.",
```

Comment:
Always on the fence on this, mentioning it anyway: should we try to hint why it is discouraged? Or say "for further information consult the upgrade guide"?